### ISSUE TITLE
1.6.4 build failure The constructor ‘PortRange’ should have 3 arguments, but has been given 2

### ISSUE THREAD

**Issue by ilovezfs** [hadolint/hadolint#195]
- [x] This is a bug report

### Expected behavior
builds
### Actual behavior
does not build
### Steps to reproduce the behavior

try to build

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
bash-3.2$ hadolint --version
bash: hadolint: command not found
```

**Dockerfile** (if relevant)

not

**Additional environment details (OS, stack version, etc.)**
macOS
GHC 8.2.2

```
==> cabal install --jobs=8 --max-backjumps=100000 --prefix=/usr/local/Cellar/hadolint/1.6.4
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/opt/ghc@8.2/lib'
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Resolving dependencies...
Notice: installing into a sandbox located at
/private/tmp/hadolint-20180422-83568-sl02hh/hadolint-1.6.4/.cabal-sandbox
Configuring hadolint-1.6.4...
Building hadolint-1.6.4...
Failed to install hadolint-1.6.4
Build log ( /private/tmp/hadolint-20180422-83568-sl02hh/hadolint-1.6.4/.cabal-sandbox/logs/ghc-8.2.2/hadolint-1.6.4-9Rm38n9WDSY9gADvYg9QjG.log ):
cabal: Entering directory '.'
Configuring hadolint-1.6.4...
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/opt/ghc@8.2/lib'
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Preprocessing library for hadolint-1.6.4..
Building library for hadolint-1.6.4..
[1 of 8] Compiling Hadolint.Bash    ( src/Hadolint/Bash.hs, dist/dist-sandbox-c439c74e/build/Hadolint/Bash.o )
[2 of 8] Compiling Hadolint.Rules   ( src/Hadolint/Rules.hs, dist/dist-sandbox-c439c74e/build/Hadolint/Rules.o )

src/Hadolint/Rules.hs:456:41: error:
    • The constructor ‘PortRange’ should have 3 arguments, but has been given 2
    • In the pattern: PortRange l m
      In a stmt of a list comprehension: PortRange l m <- ports
      In the first argument of ‘and’, namely
        ‘[l <= 65535 && m <= 65535 | PortRange l m <- ports]’
    |
456 |         and [l <= 65535 && m <= 65535 | PortRange l m <- ports]
    |                                         ^^^^^^^^^^^^^
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
hadolint-1.6.4-9Rm38n9WDSY9gADvYg9QjG failed during the building phase. The
exception was:
ExitFailure 1

/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: cabal install --jobs=8 --max-backjumps=100000 --prefix=/usr/local/Cellar/hadolint/1.6.4
```
build log https://gist.github.com/ilovezfs/a5f1277866694b72c29d9777504f317a

CC @lorenzo @zemanlx

Ref: https://github.com/Homebrew/homebrew-core/pull/26933
https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/22724/

**Comment by ilovezfs** [comment]
I can hack around the issue by replacing
```
    inreplace "package.yaml", "language-docker >=3.0.0", "language-docker ==3.0.1"
```
So likely an issue with the newer language-docker versions. Not sure whether the problem is >=4 or >=5.

**Comment by ilovezfs** [comment]
OK, building with language-docker 4.0.1 is OK, so the problem arises with language-docker 5.0.0.

**Comment by lorenzo** [comment]
We currently only support building hadolint with the stack tool, and not straight with cabal. Please use stack and the latest master code in this repo.

**Comment by lorenzo** [comment]
You are also welcome to download one of our pre-compiled binaries if you just want to use this tool instead of using it as a library or want to modify it.

**Comment by ilovezfs** [comment]
Sorry in Homebrew we only build with cabal, so that won't be an option. Also, all formula must build from source.

**Comment by ilovezfs** [comment]
Your package.yaml needs the following change
```
diff --git a/package.yaml b/package.yaml
index 06c8413..9254ce0 100644
--- a/package.yaml
+++ b/package.yaml
@@ -13,7 +13,7 @@ extra-source-files: README.md
 dependencies:
   - base >=4.8 && <5
   - parsec >=3.1
-  - language-docker >=3.0.0
+  - language-docker >=3.0.0 && < 5
 library:
   source-dirs: src
   exposed-modules:
```

**Comment by lorenzo** [comment]
@ilovezfs sorry, I forgot I had not merged a pull request yesterday. I'm in the process of releasing a new version. I'll let you know when the process is done

**Comment by ilovezfs** [comment]
@lorenzo no worries.

**Comment by ilovezfs** [comment]
@lorenzo thanks for the quick fix! Voilà: https://github.com/Homebrew/homebrew-core/pull/26943
